provider/datagen/src/bin/datagen/mod.rs

```diff
@@ -118,9 +120,9 @@ fn main() -> eyre::Result<()> {
         let locale_families = match locales_intermediate {
             AllLocales => eyre::bail!("--without-fallback needs an explicit locale list"),
             LanguageIdentifiers(lids) => lids,
-            LocaleFamilies(lfs) => lfs
+            Strings(strings) => strings
                 .into_iter()
-                .map(|family| family.write_to_string().parse().wrap_err(family))
+                .map(|s| s.parse().wrap_err_with(|| s.to_string()))
```

Comment:
nit: if the client provides multiple invalid strings, the hashset makes the produced error non-deterministic.